### PR TITLE
Xcube98 dzelge conda package deploy

### PR DESCRIPTION
**Pull request by dzelge**
This PR will introduce a system to:

- build a conda package for Python 3.6 as well as 3.7, win-64, osx-64, linux-64
- build a basic docker container on quay.io
- upload the conda packages to teh anaconda cloud
- uploading conda packages and building docker images on quay are only deployed on tags
- travis runs are not restricted

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  - conda update -q conda
+  - conda info -a
+  - conda install -c anaconda conda-verify conda-build anaconda-client
+  - curl -O https://raw.githubusercontent.com/lkiesow/automated-quay.io-deployment/master/deploy-on-quay
+
+script:
+  - conda build -c conda-forge -c default xcube
+  - . convert_builts.sh
+
+#before_deploy:
```
Comment (forman) [inline .travis.yml:24]: Why is the following commented out?
* If it is not needed, delete it.
* If it is not needed yet, comment why it is here.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    script: bash upload_to_anaconda_cloud.sh xcube
+    skip_cleanup: true
+    on:
+      tags: true
+  - provider: script
+    script: sh deploy-on-quay
+    skip_cleanup: true
+    on:
+      tags: true
+#  - provider: releases
```
Comment (forman) [inline .travis.yml:48]: Same here.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,97 @@
...
+  - user docker image ```quay.io/dzelge/xcube:[version]``` e.g.  ```docker pull quay.io/dzelge/xcube:[version]```
+
+TODO: Change the quay repo and travis accounts and setup the deploy key accordingly  
+
+    
+## Usage for conda images
+
+```bash
+
+conda install -c bc-dev xcube[=version]
```
Comment (forman) [inline README.md:37]: What would be the effort to use `-c conda-forge`?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,97 @@
...
+conda create pn xcube -c bc-dev xcube[=version]
+
+```
+
+Or use specify the xcube package in an ```environment.yml```.
+
+
+## Usage for Docker images
+
+TODO: Describe how to not be a root user in the docker container
```
Comment (forman) [inline README.md:48]: Why is this important here? Describe please.
If it really is important, then remove TODO and do provide description.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,97 @@
...
+- conda
+  - packages deployed to linux-64, win-64, osx-64 using conda convert
+- [anaconda.org](https://anaconda.org/bc-dev/xcube)
+- [travis-ci.org](https://travis-ci.com/dcs4cop/xcube-conda) 
+- docker / docker-compose
+- [quay.io](https://quay.io/repository/bcdev/xcube)
+
+## Dependencies
+
+- Can be run using python 3.6 and 3.7 and numpy 1.16
```
Comment (forman) [inline README.md:94]: Why numpy?

**Review by forman — CHANGES_REQUESTED** [review]
Helge, lets go though this side by side. I need to better understand, what's going on here.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,97 @@
+[![Docker Repository on Quay](https://quay.io/repository/bcdev/xcube/status "Docker Repository on Quay")](https://quay.io/repository/bcdev/xcube)
+[![Build Status](https://travis-ci.com/dcs4cop/xcube-conda.svg?branch=master)](https://travis-ci.com/dcs4cop/xcube-conda)
+[![Anaconda-Server Badge](https://anaconda.org/bc-dev/xcube/badges/latest_release_date.svg)](https://anaconda.org/bc-dev/xcube)
+[![Anaconda-Server Badge](https://anaconda.org/bc-dev/xcube/badges/platforms.svg)](https://anaconda.org/bc-dev/xcube)
+
 # xcube-conda
 Repository for deploying xcube as a conda package
+
+## Description
+This repository combines a conda and docker build as well as their deployment.
```
Comment (forman) [inline README.md:10]: Why conda *and* docker?

File: README.md (hunk 3 above)
Comment (dzelge) [reply README.md:37]: Nicht viel. Man muss das meta.yaml anpassen und im eigenen repo clone ein upstream hinzufügen, weil der conda build ein github repo im der conda-forge organisation anlegt.

**Comment by dzelge** [comment]
Closing due to the requirement to publish xcube on conda-forge. That changes the whole deployment process

**Pull request closed without merging**